script.metadata.editor/resources/lib/rating_updater.py

```diff
@@ -0,0 +1,467 @@
+
+        # set MPAA based on setting
+        if not ADDON.getSettingBool('skip_mpaa'):
+            if self.tmdb_type == 'movie':
+                release_dates = result['release_dates']['results']
+
+                for country in release_dates:
+                    if country.get('iso_3166_1') == country_code:
+                        for item in country['release_dates']:
+                            if item.get('certification'):
+                                self.tmdb_mpaa = item.get('certification')
+                                break
+                        break
+
+                    elif country.get('iso_3166_1') == 'US':
+                        for item in country['release_dates']:
+                            if item.get('certification'):
+                                self.tmdb_mpaa = item.get('certification')
+                                break
+
+            if self.tmdb_type == 'tv':
+                content_ratings = result['content_ratings']['results']
+
+                for country in content_ratings:
+                    if country.get('iso_3166_1') == country_code:
+                        self.tmdb_mpaa = country.get('rating')
+                        break
+
+                    elif country.get('iso_3166_1') == 'US':
+                        self.tmdb_mpaa_fallback = country.get('rating')
+
+            if self.tmdb_mpaa:
+                if country_code == 'DE':
+                    self.tmdb_mpaa = 'FSK ' + self.tmdb_mpaa
+
+                self._set_value('mpaa', self.tmdb_mpaa)
+
+            elif self.tmdb_mpaa_fallback:
+                self._set_value('mpaa', self.tmdb_mpaa_fallback)
+
+        # set IMDb ID if not available in the library
+        if not self.imdb:
+            if self.tmdb_type == 'movie':
+                self.imdb = result.get('imdb_id')
+
+            elif self.tmdb_type == 'tv':
+                self.imdb = result['external_ids'].get('imdb_id')
+
+            if self.imdb:
+                self._update_uniqueid_dict('imdb', self.imdb)
+
+        # add TVDb ID to uniqueid if missing
+        if not self.tvdb and self.tmdb_type == 'tv':
+            self.tvdb = result['external_ids'].get('tvdb_id')
+
+            if self.tvdb:
+                self._update_uniqueid_dict('tvdb', self.tvdb)
+
+    def get_tmdb_externalid(self,external_id):
+        result = tmdb_call(action='find',
+                           call=str(external_id),
+                           params={'external_source': 'imdb_id' if external_id.startswith('tt') else 'tvdb_id'}
+                           )
+
+        try:
+            if self.dbtype == 'movie':
+                self.tmdb = result['movie_results'][0].get('id')
+
+            elif self.dbtype == 'tvshow':
+                self.tmdb = result['tv_results'][0].get('id')
+
+            if self.tmdb:
+                self._update_uniqueid_dict('tmdb', self.tmdb)
+
+        except Exception:
+            pass
```

Comment:
The only thing that could happen is that there are no TV shows or movies stored. A common Kodi setup has both so I just used that way, because it's faster than doing a few if checks.

Reply:
But building logic upon exceptions is not very wise. But hey, it's a suggestion.